File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/MaodeObra.cpp
#include "../Hpp/MaodeObra.hpp"

MaoDeObra :: MaoDeObra(string nome, double valor) : nome(nome), valor(valor){}

MaoDeObra :: ~MaoDeObra() {}

// Gets

string MaoDeObra :: getNome(){
    return nome;
}

double MaoDeObra :: getValor(){
    return valor;
}

// Sets

void MaoDeObra :: setNome(string nome) {
    this->nome = nome;
}

void MaoDeObra :: setValor(double valor) {
    this->valor = valor;
}

File: materias ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Servicos/MaoDeObra.java
package sistemaoficina.Servicos;

public class MaoDeObra{
    private String nome;
    private double valor;
	
    public MaoDeObra(){}
    public MaoDeObra(String nome, double valor){
	setNome(nome);
	setValor(valor);
    }
	
    //Getters e Setters
	
    //Get e set nome
    public void setNome(String nome){
	this.nome = nome;
    }
    public String getNome(){
	return nome;
    }
	
    //Get e set valor
    public void setValor(double valor){
	this.valor = valor;
    }
	
    public double getValor(){
        return valor;
    }
	
    @Override
    public Object clone(){
        try{                
           return super.clone();
                
        }catch(CloneNotSupportedException e){
            return null;
        }
   }
}